homework/tests/example.spec.js

```diff
@@ -14,18 +14,211 @@ const data = [
+  //Check button clear
+  test.describe(`${version}: Clear button`, () => {
+    checkClearButton(version);
+  })
+
+  
+  
+  //Only field integer toggle test
+  test.describe(`${version}: "Only integer" error`, () => {
+    onlyIntegerTest(version);
+  });
+  
+  
+  //Calculations test
+  actionLabel.forEach( label =>{
+
+    test.describe(`${version}: ${label}`, () => {
+
+      //Going through data set array
+      for( let i = 0; i < numberSet.length; i++){
+        let expectedRes = returnExpectedValue(label,numberSet[i][0],numberSet[i][1]);
+        calculationTest(version,numberSet[i][0],numberSet[i][1],expectedRes,label);
+      }
+
     });
+
   });
-});
\ No newline at end of file
+
+})
+
+
+
+
+//FunctionTemplate
+function calculationTest(version, data1, data2, expectedResult, actionLabel){
+  test(actionLabel + ' '+ data1 + ' '+ data2 + " results in " + expectedResult, async ({ page }) => {
+
+    data1 = data1.toString();
+    data2 = data2.toString();
+
+    await page.goto('https://testsheepnz.github.io/BasicCalculator');
+    await page.selectOption('#selectBuild', { label: version});
+    await page.locator('#number1Field').type(data1);
+    await page.locator('#number2Field').type(data2);
+    await page.selectOption('#selectOperationDropdown', {label: actionLabel});
+    await page.locator('#calculateButton').click();
+    await expect(page.locator('#numberAnswerField')).toHaveValue(expectedResult);
+    })  
+  };
+
+function onlyIntegerTest(version){
+  test('Only integer error message appeared', async ({ page }) => {
+        
+        //Skipping action 'Concatenate'
+        for( let i = 1; i < actionLabel.length; i++){
+
+          await page.goto('https://testsheepnz.github.io/BasicCalculator');
+          await page.selectOption('#selectBuild', { label: version});
+          await page.locator('#number1Field').type('a');
+          await page.locator('#number2Field').type('b');
+          await page.selectOption('#selectOperationDropdown', {label: actionLabel[i]});
+          await page.locator('#calculateButton').click();
+
+          await page.locator('text=Number 1 is not a number');
+          //Check if error message not empty
+
+          //Does not work by label text
+          //await expect(page.locator('#errorMsgField')).toHaveValue("Number 1 is not a number");
+
+
+          //Does not work checking form text
+          //await expect(page.locator('#errorForm')).toHaveValue("Number 1 is not a number");
+
+
+          //Thorws error: expected to fail but passed
+          /*
+          const errorText = page.locator('#errorMsgField').inputValue;
+          if(String(errorText).match('not a number') === null){ //if no matching char found
+            console.log("Error messages found!\n");
+            test.fail();
+          }
+        */
+
+    }}) 
+      
+    //Integer radio button must be checked when arithmetic calculations are performed
```

Comment:
'Integers only' is a checkbox. Also I'd say the fact checkbox is selected doesn't mean it works, so would be nice to update this test to verify it :)